pkg/types/encoder.go

```diff
@@ -23,3 +23,26 @@ func YAMLEncoder(writer io.Writer, v interface{}) error {
 	_, err = writer.Write(buf)
 	return err
 }
+
+func JSONLinesEncoder(writer io.Writer, v interface{}) error {
+	if collection, ok := v.(*GenericCollection); ok {
+		encoder := json.NewEncoder(writer)
+
+		// encode the top level object first
+		err := encoder.Encode(collection.Collection)
+		if err != nil {
+			return err
+		}
+
+		// write collection objects 1 at a time
+		for _, obj := range collection.Data {
+			err = encoder.Encode(obj)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	// if we receive a type that is not a collection fall back to standard json encoding
+	return json.NewEncoder(writer).Encode(v)
```

Comment:
Am I missing something or is an `else { ... }` missing here and the entire collection and all it's data is getting sent again at the end, which goes a long way to explaining why jsonl isn't faster...

Reply:
Sure looks like it's writing everything twice.